internal/driver/flamegraph.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package driver
+
+import (
+	"encoding/json"
+	"html/template"
+	"net/http"
+
+	"github.com/google/pprof/internal/report"
+)
+
+type flameGraphNode struct {
+	Name     string
+	Value    int64
+	Children map[string]*flameGraphNode
+}
+
+func (n *flameGraphNode) add(stackPtr *[]string, index int, value int64) {
```

Comment:
Could avoid recursion here I think?

Reply:
Possibly, I just found the recursive version a bit easier to read.